go/integTest/parallelized_test.go

```diff
@@ -0,0 +1,21 @@
+// Copyright Valkey GLIDE Project Contributors - SPDX Identifier: Apache-2.0
+
+package integTest
+
+import (
+	"runtime"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/valkey-io/valkey-glide/go/api"
+)
+
+func (suite *GlideTestSuite) TestParallelizedSetWithGC() {
+	// The insane 640 parallelism is required to reproduce https://github.com/valkey-io/valkey-glide/issues/3207.
+	suite.runParallelizedWithDefaultClients(640, 640000, time.Minute, func(client api.BaseClient) {
+		runtime.GC()
+		key := uuid.New().String()
+		value := uuid.New().String()
+		suite.verifyOK(client.Set(key, value))
```

Comment:
Just to double check that this test achieve its goal without `-race` flag.

Reply:
It also catches the same mistake I made with `cgo.Handle`.
<img width="679" alt="image" src="https://github.com/user-attachments/assets/8f1425b7-5a94-4bbd-b283-f5ae14a2fbde" />

I was doing the following, which basically copied from the `cgo.Handle` example:

```go
func (p *pinner) Pin(v unsafe.Pointer) unsafe.Pointer {
	p.h = cgo.NewHandle(v)
	return unsafe.Pointer(&p.h)
}
```
But the &p.h suffers from the same problem as the payload channel. I have updated a corrected version:
```go
func (p *pinner) Pin(v unsafe.Pointer) unsafe.Pointer {
	p.h = cgo.NewHandle(v)
	return unsafe.Pointer(p.h) // Note that unsafe.Pointer(&p.h) is incorrect.
}
```

Reply:
However, `unsafe.Pointer(p.h)` is not safe either, since a `cgo.Handle` is generated by an atomic counter and is not a valid memory address. Converting to an `unsafe.Pointer` could lead Go runtime to do some wrong things.

Therefore, in the latest version, I changed the pinner interface to return an `uintptr` instead and also updated callback signatures accordingly. https://github.com/valkey-io/valkey-glide/pull/3208/commits/f4529b9131714f88ddcb17634c2dbffc80157b9f

Reply:
Given that all the pinner functions get inlined, I think it is safe to do `unsafe.Pointer(p.h)` because it will be cast to uintptr immediately. 
<img width="576" alt="image" src="https://github.com/user-attachments/assets/d856ccb8-5fdc-4243-b8c2-18881dc81802" />

Reply:
So I think this PR is ready for review again. @Yury-Fridlyand, @yipin-chen, @omanges, and @omangesg Please take a look. Thanks.